dids/did_dht_test.go

```diff
@@ -0,0 +1,170 @@
+package dids
```

Comment:
what are your thoughts on `package dids_test` vs `package dids` for tests? i'm guessing it has to do with whether we're testing non-exported methods?

Reply:
that and accessing internals of some structs to assert that they get the right state. Sometimes when theres a good amount of functionality that's not exposed, being able to access some internals, also mock internal interfaces to shortcut a bit on testing setup and test different behaviors it helps to have access to all of that